src/control/cmd/daos/pool.go

```diff
@@ -89,61 +89,93 @@ func (cmd *poolBaseCmd) PoolID() ui.LabelOrUUIDFlag {
 	return cmd.Args.Pool.LabelOrUUIDFlag
 }
 
+// poolConnect is a convenience wrapper around poolConnectAPI.
+func poolConnect(poolID, sysName string, flags uint, query bool) (C.daos_handle_t, *C.daos_pool_info_t, error) {
+	var cSysName *C.char
+	if sysName != "" {
+		cSysName = C.CString(sysName)
+		defer freeString(cSysName)
+	}
+
+	cPoolID := C.CString(poolID)
+	defer freeString(cPoolID)
+
+	var hdl C.daos_handle_t
+	var infoPtr *C.daos_pool_info_t
+	if query {
+		infoPtr = &C.daos_pool_info_t{
+			pi_bits: C.ulong(daos.DefaultPoolQueryMask),
+		}
+	}
+
+	return hdl, infoPtr, poolConnectAPI(cPoolID, cSysName, C.uint(flags), &hdl, infoPtr)
+}
+
+// poolConnectAPI is a lower-level wrapper around daos_pool_connect().
+func poolConnectAPI(poolID, sysName *C.char, flags C.uint, hdl *C.daos_handle_t, info *C.daos_pool_info_t) error {
+	return daosError(C.daos_pool_connect(poolID, sysName, flags, hdl, info, nil))
+}
+
+// poolDisconnectAPI is a convenience wrapper around daos_pool_disconnect().
+func poolDisconnectAPI(hdl C.daos_handle_t) error {
+	// Hack for NLT fault injection testing: If the rc
+	// is -DER_NOMEM, retry once in order to actually
+	// shut down and release resources.
+	rc := C.daos_pool_disconnect(hdl, nil)
+	if rc == -C.DER_NOMEM {
+		rc = C.daos_pool_disconnect(hdl, nil)
+		// DAOS-8866, daos_pool_disconnect() might have failed, but worked anyway.
+		if rc == -C.DER_NO_HDL {
+			rc = -C.DER_SUCCESS
+		}
+	}
+
+	return daosError(rc)
+}
+
 func (cmd *poolBaseCmd) connectPool(flags C.uint) error {
 	sysName := cmd.SysName
 	var cSysName *C.char
 	if sysName != "" {
 		cSysName := C.CString(sysName)
 		defer freeString(cSysName)
 	}
-	var rc C.int
+
 	switch {
 	case cmd.PoolID().HasLabel():
 		var poolInfo C.daos_pool_info_t
 		cLabel := C.CString(cmd.PoolID().Label)
 		defer freeString(cLabel)
 
 		cmd.Debugf("connecting to pool: %s", cmd.PoolID().Label)
-		rc = C.daos_pool_connect(cLabel, cSysName, flags,
-			&cmd.cPoolHandle, &poolInfo, nil)
-		if rc == 0 {
-			var err error
-			cmd.poolUUID, err = uuidFromC(poolInfo.pi_uuid)
-			if err != nil {
-				cmd.disconnectPool()
-				return err
-			}
+		if err := poolConnectAPI(cLabel, cSysName, flags, &cmd.cPoolHandle, &poolInfo); err != nil {
+			return err
+		}
+		var err error
+		cmd.poolUUID, err = uuidFromC(poolInfo.pi_uuid)
+		if err != nil {
+			cmd.disconnectPool()
+			return err
 		}
 	case cmd.PoolID().HasUUID():
 		cmd.poolUUID = cmd.PoolID().UUID
 		cmd.Debugf("connecting to pool: %s", cmd.poolUUID)
 		cUUIDstr := C.CString(cmd.poolUUID.String())
```

Comment:
nit: cUUIDstr isn't really consistent with the naming conventions used in our code base, cUUIDStr or cUuidStr would be